src/hotspot/share/runtime/mutex.cpp

```diff
@@ -419,23 +419,6 @@ bool Mutex::contains(Mutex* locks, Mutex* lock) {
   return false;
 }
 
-// NSV implied with locking allow_vm_block or !safepoint_check locks.
-void Mutex::no_safepoint_verifier(Thread* thread, bool enable) {
```

Comment:
Wouldn't it have been simpler to add the is_Java_thread() check in here rather than "inlining" this at the call-sites?